### DISCUSSION TITLE
Add Support For PostgreSQL "ANY" In ColumnTrait

### DISCUSSION THREAD

**Discussion by anshap1719** [SeaQL/sea-orm#1804] [Ideas]
#### Schema
```rust
#[derive(
    // ...
)]
#[sea_orm(table_name = "recipe")]
struct Recipe {
    #[sea_orm(primary_key, auto_increment = false)]
    id: i32,
    ingredients: Vec<String>,
}
```

Since this is a simple enough query, there really shouldn't be a need for using `Expr::cust` for this, which is what I presently have to use to achieve the following query:

`SELECT * FROM recipe WHERE 'Cheese' = ANY("ingredients");`

Adding support for an `eq_any` method in the trait behind the postgres feature flag should be easy. I'd be happy to create a PR if this is something that you are interested in supporting.

## Proposed Solutions

Add new method to `ColumnTrait` with following signature:

`fn eq_any<V>(&self, value: V) -> SimpleExpr where V: Into<Value>`

This will allow the query to look like this:

`Entity::find().filter(Column::Ingredients.eq_any("Cheese"))`

## Current Workarounds

Currently, the same query has to be implemented like this:

`Entity::find().filter(Expr::cust("'Cheese' = ANY(\"ingredients\")"))`

**Comment by anshap1719** [comment]
Hi @billy1624, Any opinions?

**Comment by tyt2y3** [comment]
The example you provided is same as `IN (...)`. However, `ANY` could be a lot more different, it can be binded to a Postgres array.

**Reply by anshap1719** [reply]
@tyt2y3 I beg to differ. My example is basically a Postgres array contains operation, which AFAIK is not achievable using `IN (...)`. That is the exact opposite of my example.

**Reply by tyt2y3** [reply]
Can you amend your post with the example Entity or Schema?

**Reply by anshap1719** [reply]
@tyt2y3 Right. My bad, I'll add it.

**Reply by anshap1719** [reply]
@tyt2y3 Added.

**Comment by tyt2y3** [comment]
There is some syntax support for the ANY operator https://docs.rs/sea-query/latest/sea_query/extension/postgres/struct.PgFunc.html#method.any

**Reply by anshap1719** [reply]
Perfect. This might be better to use in the implementation instead of Expr::cust I believe.
